draft-ietf-quic-transport.md

```diff
@@ -666,64 +666,226 @@ Version negotiation uses unprotected data.  The result of the negotiation MUST
+  32-bit integer.
 
-* CFCW: Connection Flow Control Window.  The connection level flow
-  control byte offset advertised by the sender of this parameter.
+ICSL (0x0003):
 
-* MSPC: Maximum number of incoming streams per connection.
+: The idle timeout is a value in seconds that is encoded as an unsigned 16-bit
+  integer.  The maximum value is 600 seconds (10 minutes).
 
-* ICSL: Idle timeout in seconds.  The maximum value is 600 seconds (10 minutes).
+An endpoint MAY use the following transport parameters:
 
-#### Optional Transport Parameters
+TCID (0x0004):
 
-* TCID: Indicates support for truncated Connection IDs.  If sent by a peer,
-  indicates that connection IDs sent to the peer should be truncated to 0 bytes.
-  This is expected to commonly be used by an endpoint where the 5-tuple is
-  sufficient to identify a connection.  For instance, if the 5-tuple is unique
-  at the client, the client MAY send a TCID parameter to the server.  When a
-  TCID parameter is received, an endpoint MAY choose to not send the connection
-  ID on subsequent packets.
+: The truncated connection identifier parameter indicates that packets sent to
+  the peer can omit the connection ID.  This can be used by an endpoint where
+  the 5-tuple is sufficient to identify a connection.  This parameter is zero
+  length.
 
-* COPT: Connection Options are a repeated tag field.  The field contains any
+COPT (0x0005):
+
+: Connection options are a repeated parameter.  The parameter contains any
   connection options being requested by the client or server.  These are
   typically used for experimentation and will evolve over time.  Example use
   cases include changing congestion control algorithms and parameters such as
   initial window.  (TODO: List connection options.)
 
-### Proof of Source Address Ownership
+### Values of Transport Parameters for 0-RTT
+
+Transport parameters from the server are remembered by the client for 0-RTT
+connections.  If values change as a result of completing the handshake, the
+client is expected to respect the new values.  This introduces some potential
+problems, particularly with respect to transport parameters that establish
+limits:
+
+* A client might exceed a newly declared connection or stream flow control limit
+  with 0-RTT data.  If this occurs, the client ceases transmission as though the
+  flow control limit was reached.  Once WINDOW_UPDATE frames indicating an
+  increase to the affected flow control offsets is received, the client can
+  recommence sending.
+
+* Similarly, a client might exceed the concurrent stream limit declared by the
+  server.  A client MUST reset any streams that exceed this connection.  A
+  server MAY reset these streams.
+
+A client cannot use or rely upon any other transport parameter that was enabled
+in a previous connection.  This includes those defined in this document.  A
+client MUST assume that the transport parameter was absent, unless the
```

Comment:
Are you saying that it's ok for the client to cache and use flow control limits and stream limits from a previous connection, but not other params? That seems reasonable to me, but I'd make that more explicit.